src/main/kotlin/com/coder/jetbrains/matcher/PortMatcher.kt

```diff
@@ -0,0 +1,62 @@
+package com.coder.jetbrains.matcher
+
+class PortMatcher(private val rule: String) {
+    private sealed class MatchRule {
+        data class SinglePort(val port: Int) : MatchRule()
+        data class PortRange(val start: Int, val end: Int) : MatchRule()
+        data class RegexPort(val pattern: Regex) : MatchRule()
+    }
+
+    private val parsedRule: MatchRule
+
+    init {
+        parsedRule = parseRule(rule)
+    }
+
+    fun matches(port: Int): Boolean {
+        return when (parsedRule) {
+            is MatchRule.SinglePort -> port == parsedRule.port
+            is MatchRule.PortRange -> port in parsedRule.start..parsedRule.end
+            is MatchRule.RegexPort -> parsedRule.pattern.matches(port.toString())
+        }
+    }
+
+    private fun parseRule(rule: String): MatchRule {
+        // Remove host part if present (e.g., "localhost:3000" -> "3000")
+        val portPart = rule.substringAfter(':').takeIf { ':' in rule } ?: rule
+
+        return when {
+            // Try parsing as single port
+            portPart.all { it.isDigit() } -> {
+                val port = portPart.toInt()
+                validatePort(port)
+                MatchRule.SinglePort(port)
+            }
+            // Try parsing as port range (e.g., "40000-55000")
+            portPart.matches("^\\d+-\\d+$".toRegex()) -> {
```

Comment:
this regex is quite strict, it doesn't allow any space betwen the  numbers which is an easy mistake. I personally have the bad (?) habit of "space invading" :) I could easily write something like: localhost:4000 - 5000. Do we want to be more lenient here?

Reply:
Not sure if the reference implementation allows this, but I've relaxed the regexp to allow spaces (I don't think it hurts).